lapack/gonum/dlapmr.go

```diff
@@ -0,0 +1,85 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import "gonum.org/v1/gonum/blas/blas64"
+
+// Dlapmr rearranges the rows of the m×n matrix X as specified by the permutation
+// k[0],k[1],...,k[m-1] of the integers 0,...,m-1.
+//
+// If forward is true, a forward permutation is applied:
+//  X[k[i],0:n] is moved to X[i,0:n] for i=0,1,...,m-1.
+// If forward is false, a backward permutation is applied:
+//  X[i,0:n] is moved to X[k[i],0:n] for i=0,1,...,m-1.
+//
+// k must have length m, otherwise Dlapmr will panic.
+func (impl Implementation) Dlapmr(forward bool, m, n int, x []float64, ldx int, k []int) {
+	switch {
+	case m < 0:
+		panic(mLT0)
+	case n < 0:
+		panic(nLT0)
+	case ldx < max(1, n):
+		panic(badLdX)
+	}
+
+	// Quick return if possible.
+	if m == 0 || n == 0 {
+		return
+	}
```

Comment:
Move this below the parameter checks and merge with the other quick return?

Reply:
The length check doesn't work if m==0, so this quick return has to be here. Moving the other quick return here wouldn't check that k has the right length, so I'll just leave it as is.